DEVPROD-8871

dalyd opened the pull request:
Draft PR for piping new `reference_version_id` information through the system. This has to go through several repos. There will also be matching DSI changes. I am confident of my ability to route the new field into cedar_report.json once we have these other parts working. I'm asking for early feedback about
1. What am I missing? 
2. How do we properly test these changes?

Linked PRs
- [Poplar](https://github.com/evergreen-ci/poplar/pull/149/files)
- [SPS](https://github.com/10gen/signal-processing-service/pull/843/files) 
- [Juniper](https://github.com/evergreen-ci/juniper/pull/4/files)

dalyd commented on model/perf.go:
```diff
@@ -334,39 +334,41 @@ func (result *PerformanceResult) Close(ctx context.Context, completedAt time.Tim
+	Mainline           bool                 `bson:"mainline"`
+	OverrideInfo       OverrideInfo         `bson:"override_info,omitempty"`
+	Schema             int                  `bson:"schema,omitempty"`
+        ReferenceVersionID string               `bson:"version,omitempty"`
```
This line is the change. The rest is formatting.

dalyd commented on model/perf.go:
```diff
@@ -334,39 +334,41 @@ func (result *PerformanceResult) Close(ctx context.Context, completedAt time.Tim
+	perfResultInfoMainlineKey        = bsonutil.MustHaveTag(PerformanceResultInfo{}, "Mainline")
+	perfResultInfoOverrideInfoKey    = bsonutil.MustHaveTag(PerformanceResultInfo{}, "OverrideInfo")
+	perfResultInfoSchemaKey          = bsonutil.MustHaveTag(PerformanceResultInfo{}, "Schema")
+        perfResultInfoReferenceVersionID = bsonutil.MustHaveTag(PerformanceResultInfo{}, "ReferenceVersionID")
```
This last line is the change.  The rest is formatting.

zamj commented:
Going off of @SimonEismann 's PR, I believe this is all that's necessary for the new field to begin to be read out of the database. However, I believe more needs to be done on the Poplar and Cedar side of things. Poplar sends that json data over a grpc call to Cedar.

[Cedar GRPC function](https://github.com/evergreen-ci/cedar/blob/DEVPROD-7875/rpc/internal/perf_service.go#L45)
[Cedar JSON struct](https://github.com/evergreen-ci/cedar/blob/DEVPROD-7875/rpc/internal/perf.pb.go#L310)
[Poplar calling that GRPC function](https://github.com/evergreen-ci/poplar/blob/b43ddb865c3f9e10266aba7540b9426119a37a82/rpc/client.go#L165)
[Parent Poplar Function that is called by Evergreen](https://github.com/evergreen-ci/poplar/blob/b43ddb865c3f9e10266aba7540b9426119a37a82/rpc/client.go#L31)
[Evergreen calling that Poplar parent function (perf.send)](https://github.com/evergreen-ci/evergreen/blob/c80fb627788477e2875e80f2195965686c39b403/agent/command/perf_send.go#L79)

dalyd commented:
Thanks Jeff. That really helped. I've updated the perf.proto, regenerated the generated file, did a couple of other changes, and opened up a [juniper PR](https://github.com/evergreen-ci/juniper/pull/4) as well. I'm not sure if that gets pulled over or not. More below in your list. I'm not sure we need changes in all the places. 

> Going off of @SimonEismann 's PR, I believe this is all that's necessary for the new field to begin to be read out of the database. However, I believe more needs to be done on the Poplar and Cedar side of things. Poplar sends that json data over a grpc call to Cedar.
> 
> That entire grpc stack needs to be updated to have this new field in it. So, the proto implementation needs to be updated, new clients and objects generated, and then both Poplar and Cedar updated to use the new generated things.
> 
[Cedar GRPC function](https://github.com/evergreen-ci/cedar/blob/DEVPROD-7875/rpc/internal/perf_service.go#L45) 
[Cedar JSON struct](https://github.com/evergreen-ci/cedar/blob/DEVPROD-7875/rpc/internal/perf.pb.go#L310) 
Updated perf.proto (in cedar and juniper) and regenerated this file. I think this takes care of this item and the previous unless I missed something. 

[Poplar calling that GRPC function](https://github.com/evergreen-ci/poplar/blob/b43ddb865c3f9e10266aba7540b9426119a37a82/rpc/client.go#L165)
I didn't update this line, but I did a little bit above this to pipe the new field into ResultData. ResultData type should be updated from the juniper update. 

 [Parent Poplar Function that is called by Evergreen](https://github.com/evergreen-ci/poplar/blob/b43ddb865c3f9e10266aba7540b9426119a37a82/rpc/client.go#L31)
I think this one is also covered by juniper updates. 

 [Evergreen calling that Poplar parent function (perf.send)](https://github.com/evergreen-ci/evergreen/blob/c80fb627788477e2875e80f2195965686c39b403/agent/command/perf_send.go#L79)
I need to update the poplar version in evergreen for this? I think that's what was said at standup. 

@zamj could you take another look and let me know where I'm wrong?
